test-data/unit/check-bytes-str-unicode-python2.test

```diff
@@ -0,0 +1,49 @@
+-- Test cases for bytes/str/unicode in Python 2.x.
+
+[case testBytesStrUnicodeBasics]
+# flags: --hide-error-context --fast-parser
+def needs_bytes(b):
+    # type: (bytes) -> None
+    needs_bytes(b)
+    needs_str(b)  # E: Argument 1 to "needs_str" has incompatible type "bytes"; expected "str"
+    needs_unicode(b)  # E: Argument 1 to "needs_unicode" has incompatible type "bytes"; expected "unicode"
+
+def needs_str(s):
+    # type: (str) -> None
+    # TODO: The following line should not be an error
```

Comment:
Note this TODO. I want mypy to auto-coerce str to either unicode or bytes depending on context.